### PR TITLE
Fix offset in rendering of `TransformBox` of image/labels layers

### LINKED ISSUE #6676: Transform interaction box is 0.5 px offset
### 🐛 Bug Report

The overlay transform interaction box is offset from the image layer data by 0.5 px.
Top left:
image
bottom right:
image

### 💡 Steps to Reproduce

1. Open any napari sample image in napari
2. Press `2` to activate Transform mode
3. Zoom in to a corner to see that the box doesn't match the edge of the data
4. `viewer.layers[0].bounding_box.visible = True` to see that the *bounding_box* is correct

### 💡 Expected Behavior

The transform interaction overlay should match the bounding box overlay.

### 🌎 Environment

napari: 0.5.0a2.dev543+gad78ee21a
Platform: macOS-13.3.1-arm64-arm-64bit
System: MacOS 13.3.1
Python: 3.11.7 | packaged by conda-forge | (main, Dec 23 2023, 14:38:07) [Clang 16.0.6 ]
Qt: 6.6.1
PyQt6: 
NumPy: 1.26.3
SciPy: 1.12.0
Dask: 2024.1.1
VisPy: 0.14.1
magicgui: 0.8.1
superqt: 0.6.1
in-n-out: 0.1.9
app-model: 0.2.4
npe2: 0.7.4

OpenGL:
- GL version: 2.1 Metal - 83.1
- MAX_TEXTURE_SIZE: 16384
- GL_MAX_3D_TEXTURE_SIZE: 2048

Screens:
- screen 1: resolution 1680x1050, scale 2.0

Settings path:
- /Users/piotrsobolewski/Library/Application Support/napari/napari-dev_a1eb8b76ba95fa16ad06e26097b46b8455dfbf0b/settings.yaml

### 💡 Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Czaki**
# References and relevant issues

closes #6676

# Description

Add offset shift for bounding box to fix bounding box position

### CODE REVIEW

**Comment by Czaki** [comment]
@psobolewskiPhD As you open Issue, could you edit the description of PR?

**Comment by psobolewskiPhD** [comment]
This fixes the transform overlay, but now the bounding box one is off by 0.5 px:
image

Maybe don't need the offset on the vispy side now? see:
https://github.com/napari/napari/issues/6676#issuecomment-1955726376

**Comment by Czaki** [comment]
fixed

**Comment by psobolewskiPhD** [comment]
Now the issue is in 3D, as that code was gated behind 2D only:
image
image

Maybe easier to just add an offset to the interaction box?

**Review by brisvag — CHANGES_REQUESTED** [review]
I don't think this is the right way to fix it. Check out #6690!

**Comment by psobolewskiPhD** [comment]
Based on https://github.com/napari/napari/issues/6676#issuecomment-1964634819 I think that #6690 makes more sense and is cleaner.

File: napari/_vispy/layers/base.py [hunk 1]
```diff
@@ -214,6 +214,8 @@ def _on_matrix_change(self):
         affine_matrix[: matrix.shape[0], : matrix.shape[1]] = matrix
         affine_matrix[-1, : len(translate)] = translate
 
+        offset = np.ones(len(self.layer._slice_input.displayed))
```
Comment (Czaki) [inline napari/_vispy/layers/base.py]:
```suggestion
        offset = np.zeros(len(self.layer._slice_input.displayed))
```

**Review by jni — APPROVED** [review]

**Review by brisvag — APPROVED** [review]
Ok, I think it makes sense now, and it's indeed cleaner than my version. LGTM!

**Comment by Czaki** [comment]
Is the title ok?

**Comment by brisvag** [comment]
Changed, this is a fix for transformbox and not boundingbox.

**Pull request merged**